src/nodekit.ts

```diff
@@ -91,25 +91,33 @@ export class NodeKit {
             utils: this.utils,
         });
 
-        this.addShutdownHandler(() => {
-            return new Promise((resolve) => {
-                this.tracer.close?.(() => resolve());
-            });
-        });
+        this.addShutdownHandler(() => new Promise<void>((resolve) => this.tracer.close(resolve)));
 
-        ['SIGTERM', 'SIGINT'].forEach((signal) => {
-            process.on(signal, () => {
-                Promise.all(this.shutdownHandlers.map((handler) => handler()))
-                    .then(() => process.exit(0))
-                    .catch((error) => {
-                        this.ctx.logError('Error executing shutdown handlers', error);
-                        process.exit(1);
-                    });
-            });
-        });
+        this.setupShutdownSignals();
     }
 
     addShutdownHandler(handler: ShutdownHandler) {
         this.shutdownHandlers.push(handler);
     }
+
+    private setupShutdownSignals() {
+        const signals = ['SIGTERM', 'SIGINT'] as const;
+
+        const handleSignal: ShutdownHandler = (signal) => {
+            signals.forEach((off) => process.off(off, handleSignal));
```

Comment:
We should remove all handlers on shutdown signal

Reply:
Why they should be removed on shutdown signal?

Reply:
```suggestion
            signals.forEach((signalName) => process.off(signalName, handleSignal));
```

Reply:
> What will happen if they won't be removed?

In such case signals may be fired 2 and more times.